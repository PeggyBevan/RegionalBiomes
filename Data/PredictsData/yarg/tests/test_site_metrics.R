TestSiteMetrics <- function() {
  # Numeric
  trait1 <- TraitDataset(reference='Test', trait='M', unit='kg', 
                         taxa=paste('Sp', 1:6), kingdoms='Animalia', 
                         ranks='Species', values=1:6)

  # Logical
  trait2 <- TraitDataset(reference='Test', trait='Specialist', unit='Logical', 
                         taxa=paste('Sp', 1:5), kingdoms='Animalia', 
                         ranks='Species', values=c(FALSE,TRUE,TRUE,TRUE,TRUE))

  # TODO Categorical

  diversity <- read.csv('test_site_metrics_input.csv')
  diversity <- AddTraits(diversity, trait1, trait2)

  # Add SS and SSS
  diversity <- AddSummaryColumns(diversity)

  # Preprocessing
  diversity <- DropInvalidMetricsAndMethods(diversity)
  diversity <- CorrectSamplingEffort(diversity)

  actual <- SiteMetrics(diversity, traits=c('M_kg', 'Specialist_Logical'))

  cat('HACK: TestSiteMetrics ignoring Richness_rarefied.\n')
  expected <- read.csv('test_site_metrics_expected.csv')
  actual <- actual[,colnames(expected)]
  AssertEqual(expected, actual)

  cat('HACK: TestSiteMetrics setting Taxon_name_entered for non-unique sites.\n')
  diversity$Taxon_name_entered <- diversity$Taxon
  actual <- SiteMetrics(diversity, traits=c('M_kg', 'Specialist_Logical'), sites.are.unique=FALSE)
  actual <- actual[,colnames(expected)]
  AssertEqual(expected, actual)

  # Can't compute site metrics of categorical traits
  categorical <- TraitDataset(reference='Test', trait='Diet', unit='Category', 
                              taxa=paste('Sp', 1:3), kingdoms='Animalia', 
                              ranks='Species', 
                              values=c('Herbivore', 'Omnivore', 'Carnivore'))
  diversity <- AddTraits(diversity, categorical)
  AssertRaises(SiteMetrics(diversity, traits='Diet_Category'))
}
