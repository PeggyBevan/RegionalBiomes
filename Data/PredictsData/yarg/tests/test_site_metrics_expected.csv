Source_ID,Study_number,Site_number,Diversity_metric_type,SS,SSS,Total_abundance,Species_richness,Simpson_diversity,ChaoR,CWM_M_kg,IPR_M_kg
A,1,1,Abundance,A 1,A 1 1,10,4,3.3333333333,4,3,3
A,1,2,Abundance,A 1,A 1 2,10,1,1,1,4,0
A,1,3,Abundance,A 1,A 1 3,20,1,1,1,5,0
A,2,1,Abundance,A 2,A 2 1,20,1,1,1,1,0
B,1,1,Occurrence,B 1,B 1 1,,1,,,1,0
B,1,2,Occurrence,B 1,B 1 2,,3,,,3,2
B,2,1,Occurrence,B 2,B 2 1,,1,,,1,0
B,2,2,Occurrence,B 2,B 2 2,,0,,,,0
C,1,1,Species richness,C 1,C 1 1,,12,,,1,0
C,1,2,Species richness,C 1,C 1 2,,20,,,1,0
C,1,3,Species richness,C 1,C 1 3,,0,,,,0
C,2,1,Species richness,C 2,C 2 1,,0,,,,0
C,2,2,Species richness,C 2,C 2 2,,5,,,1,0
C,2,3,Species richness,C 2,C 2 3,,8,,,1,0
